### DISCUSSION TITLE
Can I access the `iss` claim?

### DISCUSSION THREAD

**Discussion by marcvs** [pysnippet/fastapi-oauth2#26] [Q&A] [answered]
The user model nicely includes data from the `userinfo` endpoint in `request.user`. I do however miss the `iss` claim. Is there any way to access it?

**Comment by ArtyomVancyan** [comment, marked as answer]
The `request.user` contains the user information obtained from the particular API endpoint based on the used IDP and most likely should not contain any JWT claims such as `iss`. After the successful login, you have the `access_token` in your request [context](https://docs.pysnippet.org/fastapi-oauth2/references/tutorials#access-token) and accessible by `request.auth.provider.access_token`. Try to decode it with the algorithm it is signed with. For determining the algorithm, you can paste your token in https://jwt.io/ and also ensure that it contains the `iss` claim or not.

```python
from jose.jwt import encode as jwt_decode

@router.get("/")
async def root(request: Request):
    print(jwt_decode(request.auth.provider.access_token, "", "<alg>"))
```
